components/lvgl.rst

```diff
@@ -0,0 +1,2449 @@
+    - **grid_columns** (**Required**): The number of columns in the grid, expressed a list of values in pixels, ``CONTENT`` or ``FR(n)`` (free units, where ``n`` is a proportional integer value).
+    - **grid_row_align** (*Optional*, string): How to align the row. Works only when ``grid_rows`` is given in pixels. Possible options below.
+    - **grid_column_align** (*Optional*, string): How to align the column. Works only when ``grid_columns`` is given in pixels. Possible options below.
+    - **pad_row** (*Optional*, int16): Set the padding between the rows, in pixels.
+    - **pad_column** (*Optional*, int16): Set the padding between the columns, in pixels.
+    
+In a grid layout, *all the widgets placed on the grid* will get some additional configuration variables to help with placement:
+
+    - **grid_cell_row_pos** (**Required**, int16): Position of the widget, in which row to appear (0 based count).
+    - **grid_cell_column_pos** (**Required**, int16): Position of the widget, in which column to appear (0 based count).
+    - **grid_cell_x_align** (*Optional*, string): How to align the widget horizontally within the cell. Can also be applied through :ref:`lvgl-styling`. Possible options below.
+    - **grid_cell_y_align** (*Optional*, string): How to align the widget vertically within the cell. Can also be applied through :ref:`lvgl-styling`. Possible options below.
+    - **grid_cell_row_span**  (*Optional*, int16): How many rows to span across the widget. Defaults to ``1``.
+    - **grid_cell_column_span** (*Optional*, int16): How many columns to span across the widget. . Defaults to ``1``.
+
+    .. note::
+
+        These ``grid_cell_`` variables apply to widget configuations!
+
+Values for use with ``grid_column_align``, ``grid_row_align``, ``grid_cell_x_align``, ``grid_cell_y_align``:
+
+        - ``START``: means left horizontally and top vertically (default).
+        - ``END``: means right horizontally and bottom vertically.
+        - ``CENTER``: simply center.
+        - ``STRETCH``: stretch the widget to the cell in the respective direction. Does not apply to ``grid_column_align``, ``grid_row_align``.
+        - ``SPACE_EVENLY``: items are distributed so that the spacing between any two items (and the space to the edges) is equal.
+        - ``SPACE_AROUND``: items are evenly distributed in the track with equal space around them. Note that visually the spaces aren’t equal, since all the items have equal space on both sides. The first item will have one unit of space against the container edge, but two units of space between the next item because that next item has its own spacing that applies.
+        - ``SPACE_BETWEEN``: items are evenly distributed in the track: first item is on the start line, last item on the end line.
+
+.. tip::
+
+    To visualize real, calculated sizes of transparent widgets you can temporarily set ``outline_width: 1`` on them.
+
+.. _lvgl-widgets:
+
+Widgets
```

Comment:
I think at a minimum, this whole widgets block should be its own page/file
`components/lvgl/widgets.rst`
And rename this main one to `components/lvgl/index.rst`

> Eg. almost every widget has specific Actions/Conditions/Triggers in addition to generic ones. One clicks to compare the generic ones vs the specific ones, has to change page and navigate back-forth with the browser between two pages, whilst if they were on the same page it's just scrolling up and down or Ctrl+F...

The generic widget actions should all be included on the widgets page as that is what they relate to.

Jumping up and down a page is horrible, much easier to just open different browser tabs with the different sections/pages side-by-side.